Svc/BufferManager/BufferManagerComponentImpl.cpp

```diff
@@ -192,9 +192,9 @@ namespace Svc {
     // allocate memory
     void *memory = allocator.allocate(memId,allocatedSize,recoverable);
     // make sure the memory returns was non-zero and the size requested
-    FW_ASSERT(memory);
-    FW_ASSERT(
-      memorySize == allocatedSize,
+    FW_ASSERT(memory != nullptr && memorySize == allocatedSize,
+      mgrId, memId,
+      static_cast<FwAssertArgType>(reinterpret_cast<FwSizeType>(memory)),
```

Comment:
Why use `reinterpret_cast<FwSizeType>` on `memory` here? Are you looking to print the memory address? We have `PlatformPointerCastType` to represent pointers as integers

Reply:
I somehow didn't realize that type existed! Yes, that's definitely what I want.